ambry-cloud/src/main/java/com/github/ambry/cloud/RecoveryManager.java

```diff
@@ -107,30 +135,27 @@ public CloudToStoreReplicationManager(ReplicationConfig replicationConfig, Clust
     this.clusterParticipant = clusterParticipant;
     this.instanceNameToCloudDataNode = new AtomicReference<>(new ConcurrentHashMap<>());
     this.vcrNodes = new AtomicReference<>(new ArrayList<>());
-    this.persistor =
-        new DiskTokenPersistor(cloudReplicaTokenFileName, mountPathToPartitionInfos, replicationMetrics, clusterMap,
-            tokenHelper, storeManager);
+    this.persistor = null; // No need of a persistor
     trackPerDatacenterLagInMetric = replicationConfig.replicationTrackPerDatacenterLagFromLocal;
+    this.recoveryMetrics = new RecoveryMetrics(metricRegistry);
+    this.networkClientFactory.getNetworkClient(); // Test connection to Azure Storage and other services
   }
 
   @Override
   public void start() {
-    // Add listener for vcr instance config changes
-    vcrClusterSpectator.registerInstanceConfigChangeListener(new InstanceConfigChangeListenerImpl());
-    vcrClusterSpectator.registerLiveInstanceChangeListener(new LiveInstanceChangeListenerImpl());
-
-    // Add listener for new coming assigned partition
-    clusterParticipant.registerPartitionStateChangeListener(
-        StateModelListenerType.CloudToStoreReplicationManagerListener, new PartitionStateChangeListenerImpl());
-
-    // start background persistent thread
-    // start scheduler thread to persist index in the background
-    scheduler.scheduleAtFixedRate(persistor, replicationConfig.replicationTokenFlushDelaySeconds,
-        replicationConfig.replicationTokenFlushIntervalSeconds, TimeUnit.SECONDS);
-
+    // Use a static clustermap file on the host to know partitions assigned to the host.
+    // Assume that there is no helix/zookeeper available post disaster.
+    // In this design, server does not talk to vcr but directly to Azure.
+    this.clusterMap.getAllPartitionIds(clusterMapConfig.MAX_REPLICAS_ALL_DATACENTERS).forEach(partition -> {
```

Comment:
May be this needs to be changed to `clusterMapConfig.clusterMapDefaultPartitionClass` instead of hardcoded `clusterMapConfig.MAX_REPLICAS_ALL_DATACENTERS` ? Video cluster has default partition class configured as `video-3111`.

Also, QQ.. Would this recovery code be running on ambry server hosts? If so, would we need to filter partitions that belong to this particular host?

Reply:
Thanks, I will add the filtering. The disaster scenario assumed is that we have no helix and each server is configured using a static clustermap. 

As for why this code runs in server, there are only two places recovery code can run - server or VCR. VCR is just a big bottleneck and a proxy to Azure. It doesn't server any other purpose. A server is fully capable of directly fetching data from Azure and would be a lot faster. We discussed this in a meeting last sem.